modules/stock/commerce_pos_stock.module

```diff
@@ -41,3 +41,40 @@ function commerce_pos_stock_form_commerce_pos_return_alter(&$form, &$form_state,
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function commerce_pos_stock_form_commerce_pos_exchange_alter(&$form, &$form_state, $form_id) {
```

Comment:
Would it make sense to have this same function add this checkbox for Returns as well as Exchanges? Just wondering...